### PR TITLE
Fix exporting Sum to onnx

### PR DESCRIPTION
**Pull request by bddppq**
ONNX Sum does elementwise sum of all inputs, which is not PyTorch's sum

#7650

### CODE REVIEW

File: torch/onnx/symbolic.py [hunk 1]
```diff
@@ -216,8 +216,6 @@ def mean(g, self, dim=None, keepdim=None):
 
 
 def sum(g, self, dim=None, keepdim=None):
-    if dim is None and keepdim is None:
```
Comment (soumith) [inline torch/onnx/symbolic.py]: aren't you removing the dim=None case, which translates to Sum?

Comment (soumith) [reply torch/onnx/symbolic.py]: does `g.op("ReduceSum"` handle case when axes_i=None and does a Sum over all dimensions (and return a single number)?

Comment (bddppq) [reply torch/onnx/symbolic.py]: It should never translate to Sum. If dim is missing, then ONNX ReduceSum sums over all, which matches what PyTorch does.

**Review by soumith — APPROVED** [review]

File: torch/onnx/symbolic.py (hunk 1 above)
Comment (bddppq) [reply torch/onnx/symbolic.py]: Good point, updated to apply the fix to prod and mean

**Review by houseroad — APPROVED** [review]
LGTM

**Comment by houseroad** [comment]
Update expect files in onnx-fb-universe?

**Comment by houseroad** [comment]
Awesome!

**Pull request merged**